pymemcache/client/retrying.py

```diff
@@ -0,0 +1,48 @@
+import collections
+import functools
+import inspect
+import socket
+import time
+import logging
+
+import pymemcache.exceptions as pymemexception
+
+logger = logging.getLogger(__name__)
+
+def retry(func):
+    @functools.wraps(func)
+    def func_call(*args, **kwargs):
+        ntries=15
```

Comment:
Still WIP so hard coded, but the goal is to provide some configurable options (backoff, number of retry, duration, etc...)

Reply:
We need more robust strategy, like a backoff strategy with a range of possible value to avoid to flood the server and manage system workload more finely. If many clients try to reconnect at the same time the backoff strategy will help to spread the load.